src/aspire/volume/volume.py

```diff
@@ -176,6 +179,51 @@ def _set_symmetry_group(self, value):
             )
         self._symmetry_group = value
 
+    def _symmetry_group_warning(self):
+        """
+        Warn when an operation has the potential to change the symmetry
+        of a volume or the alignment of symmetry axes.
+        """
+        msg = (
+            f"`symmetry_group` attribute is being set to `C1`. This operation may"
+            f" effect the symmetry (or symmetric alignment) of the volume. To reset the"
+            f" symmetry group run `self._set_symmetry_group('{self.symmetry_group}')`."
+        )
+        warnings.warn(msg, UserWarning, stacklevel=4)
+
+    def _result_symmetry(self, other=None):
+        """
+        Check if `other` will alter symmetry of `self` and return resulting symmetry_group.
+
+        :return: SymmetryGroup instance.
+        """
+        # No need to check for C1.
+        if str(self.symmetry_group) == "C1":
+            return self.symmetry_group
+
+        # Handle case of transformations of `self`.
+        if other is None:
+            self._symmetry_group_warning()
+            return IdentitySymmetryGroup(dtype=self.dtype)
+
+        # Handle binary operations of `self` and `other`.
+        result_symmetry = self.symmetry_group
+
+        # Warn and set to IdentitySymmetryGroup if `other` is symmetrically incompatible.
+        if isinstance(other, Volume) and str(self.symmetry_group) != str(
```

Comment:
We can give `SymmetryGroup` classes a notion of equality.

Reply:
Added override of `__eq__` for `SymmetryGroup`s. As is, I am allowing symmetry groups of differing `dtype` to be equivalent. Can change this if you think it will be problematic.